CENNZnet_format.md

```diff
@@ -67,15 +69,17 @@ Decode steps:
         - Read the last 7 bits as `method_count`
     2. Read 32 bytes as `module_name`
     3. If `has_block_cooldown`
-        - Read 4 bytes as 'block_cooldown`
+        - Read 4 bytes as `block_cooldown`
     4. Begin reading methods list items
         1. Read the first byte
             - Read the first bit as `has_block_cooldown` Boolean
-            - Ignore 7 bits
+            - Read the remaining 7 bits as `constraints_count`
         2. Read 32 bytes as `method_name`
         3. If `has_block_cooldown`
-            - Read 4 bytes as 'block_cooldown`
-        4. Repeat until `method_count` matches iterations
+            - Read 4 bytes as `block_cooldown`
+        4. If `constraints_count` > 0
```

Comment:
if `has_constraints` bit is set
- read 1 byte as `constraints_length`
- read `constraints_length` bytes as `constraints`